packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -70,12 +71,15 @@ export class Autocomplete extends React.PureComponent {
     // Downshift's `getInputProps` method
     return React.Children.map(this.props.children, child => {
       if (isTextField(child)) {
-        return React.cloneElement(
-          child,
-          getInputProps({
-            id: this.id
-          })
-        );
+        const propOverrides = {
```

Comment:
We opted to break the child props being overridden out to their own `const` for better readability. These functions are being composed into Downshift here: https://github.com/paypal/downshift/blob/master/src/downshift.js#L609

Hat tip to @pwolfert for explaining how these are working.